worker/src/RTC/TransportCongestionControlServer.cpp

```diff
@@ -192,13 +193,23 @@ namespace RTC
 			return;
 		}
 
-		// Set base sequence num and reference time.
-		this->transportCcFeedbackPacket->SetBase(this->transportCcFeedbackWideSeqNumStart, it->second);
-
 		for (; it != this->mapPacketArrivalTimes.end(); ++it)
 		{
-			auto result =
-			  this->transportCcFeedbackPacket->AddPacket(it->first, it->second, this->maxRtcpPacketLen);
+			auto sequenceNumber = it->first;
+			auto timestamp      = it->second;
+
+			// If the base is not set in this packet let's set it.
+			// NOTE: This maybe needed many times during this loop since the current
+			// feedback packet maybe a fresh new one if the previous one was full (so
+			// already sent) or failed to be built.
+			if (!this->transportCcFeedbackPacket->IsBaseSet())
```

Comment:
IMO this logic should be inside `this->transportCcFeedbackPacket->AddPacket()` itself. So It sets the base if not set. It's a `TransportCcFeedbackPacket` responsibility.

Reply:
This is exactly how it was before the PR that introduced this regression. Please look at it.

Reply:
There is a difference. In the original code before the regression PR (where there was no SetBase()) the first packet/info passed to AddPacket() was used to set the base fields and then it returned early. In the new code (before and after this PR) the same packet/info is used to set SetBase() and is later passed to AddPacket() where it does make that method return early. Not sure about the implications.

Reply:
So there is a difference. The AddPacket() in "modern" code will NOT "ignore" the packet that was used in `SetBase()` due to this:

```c++
			// If the wide sequence number of the new packet is lower than the latest seen,
			// ignore it.
			// NOTE: Not very spec compliant but libwebrtc does it.
			// Also ignore if the sequence number matches the latest seen.
			if (!RTC::SeqManager<uint16_t>::IsSeqHigherThan(sequenceNumber, this->latestSequenceNumber))
			{
				return AddPacketResult::SUCCESS;
			}
```

because the value of `this->latestSequenceNumber` was set in `SetBase()` this way:

```c++
this->latestSequenceNumber = sequenceNumber - 1;
```

So the condition above doesn't match. In summary, in the "modern" code (after the addition of `SetBase()`) the first packet is used to set the base fields and it is ALSO processed. I don't really know why this is needed.

Reply:
Anyway I'm gonna get rid of that `SetBase()` and make the first packet givne to `AddPacket()` set the base (as before) and also be processed.

Reply:
Nope, I am wrong. And honestly I don't understand it. In the modern code this is how it works:

- `SetBase()` is called with `seqNumber: this->transportCcFeedbackWideSeqNumStart` and `timestamp:` the value of the first entry in the `mapPacketArrivalTimes` map (after doing `lower_bound(this->transportCcFeedbackWideSeqNumStart)` on it).
- And then the first time `AddPacket()` is called its arguments are `seqNumber:` the key of the first entry in the `mapPacketArrivalTimes` (after doing `lower_bound(this->transportCcFeedbackWideSeqNumStart)` on it) and `timestamp:` the value of that entry.
- This is, `timestamp` has the same value in the argument passed to `SetBase()` and in the first call to `AddPacket()`.

I'm gonna assume that there is a good rationale for this change (read https://github.com/versatica/mediasoup/issues/1059) so I won't change it.